vault/diagnose/tls_verification.go

```diff
@@ -136,11 +176,25 @@ func ParseTLSInformation(certFilePath string) ([]*x509.Certificate, []*x509.Cert
 			leafCerts = append(leafCerts, cert)
 		}
 	}
+
+	rootSubjects := GetCertsSubjects(rootCerts)
```

Comment:
I am not sure if this is needed or not! For self-signed certs, the cert can verify itself, and the subject and issuer are the same. So, just checking if we have subjects is not enough to determine if the cert is a self-signed cert. So, maybe we can just check if  "len(rootCerts) == 0".